### PR TITLE
[Feat] GeoArrow incremental rendering (1)

### PR DESCRIPTION
**Pull request by lixun910**
This PR is trying to implement progressive rendering of large GeoArrow file by batches. The idea is borrowed from vector tile rendering that each batch (like each tile) is parsed and rendered as a deck.gl geojsonlayer. 

incremental-render-utah
(The main thread is not fully blocked even though we don't implement the web worker version)

- instantaneous rendering of the first batch
- rendering while loading more batches

Note:

- Arrow: batch size can be set by -lco BATCH_SIZE=65536 (rows)

Some issues:
 - Ideally, we can parallelize the parsing and tesselation jobs, and each worker handles one batch. However, the data of each batch points to the same underneath array buffer of the same arrow table. That means we can't make it transferrable for each worker, and there seems no way to reattach the transferred buffer back to the arrow table in the main thread.
 - Alternatively, we can make a hard copy of the slice of the array buffer for each batch. However, it will be not efficient. 

Any suggestions or comments will be appreciated. Thanks!

### CODE REVIEW

File: src/layers/src/layer-utils.ts [hunk 1]
```diff
@@ -79,6 +91,68 @@ export function getGeojsonLayerMetaFromArrow({
   };
 }
 
+/*
```
Comment (ibgreen) [inline src/layers/src/layer-utils.ts:94]: Either cut or add comment explaining why we are keeping this code. Maybe this is related to the other PR?

**Review by ibgreen — APPROVED** [review]
LGTM Is this still rendering using multiple GeoJSONLayers? I don't see that in the code...

**Comment by lixun910** [comment]
I think it’s ready. Can’t wait to test it on website.

**Comment by ilyabo** [comment]
Great stuff @lixun910! 👍

**Pull request merged**